### PR TITLE
feat: adds vice.com route #15694

### LINKED ISSUE #15694: Vice.com topics
### Category

New media

### Website URL

https://www.vice.com

### Website description

vice.com a traditional media news 

### What content should be included?

latest feed of various topics

### Additional description

Creating so I can fill in the PR

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by K33k0**
## Involved Issue / 该 PR 相关 Issue

Close #15694

## Example for the Proposed Route(s) / 路由地址示例

```routes
/vice/topic/politics/en
/vice/topic/politics
/vice/topic/internet
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
New route for vice.com (seriously messed up the last PR)

### CODE REVIEW

File: lib/routes/vice/topic.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import cache from '@/utils/cache';
+
+export const route: Route = {
+    path: '/:topic/:content/:language?',
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:8]:
```suggestion
    path: '/topic/:topic/:language?',
```
prepanding `/topic` will make this route won't collide with any future routes.

File: lib/routes/vice/topic.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    name: 'Topic',
+    maintainers: ['K33k0'],
+    handler,
+    url: 'vice.com/',
+};
+
+async function handler(ctx) {
+    const language = ctx.req.param('language');
+    const topic = ctx.req.param('topic');
+    const content = ctx.req.param('content');
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:39]:
```suggestion
    const { language = 'en', topic } = ctx.req.param();
```

File: lib/routes/vice/topic.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    let items = null;
+    const response = await ofetch(`https://www.vice.com/${language ?? 'en'}/topic/${topic}`);
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:41]:
```suggestion
    const response = await ofetch(`https://www.vice.com/${language}/topic/${topic}`);
```

File: lib/routes/vice/topic.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        .map((item) => {
+            item = $(item);
+            return {
+                title: $(item).find('h3').first().text(),
+                link: `https://vice.com${$(item).find('h3 > a').first().attr('href')}`,
+                pubDate: parseDate(new Date($(item).find('time').first().attr('datetime') * 1)),
+                author: $(item).find('.vice-card-details__byline').first().text(),
+                description: $(item).find('.vice-card-dek').first().text(),
+            };
+        });
```
[the remark this answers is not in the document]
Comment (K33k0) [reply lib/routes/vice/topic.ts:54]: Nice, I did give the next_data a go but was getting malformed json data. Glad you it working

[the remark this answers is not in the document]
Comment (K33k0) [reply lib/routes/vice/topic.ts]: All now implemented, I hope I've credited you in the correct way. I've included you in the commit message as github does. Thank you

File: lib/routes/vice/topic.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                          // Select the first element with the class name 'comment-body'
+                          item.description = $('.short-form').first().html();
+
+                          // Every property of a list item defined above is reused here
+                          // and we add a new property 'description'
+                          return item;
+                      })
+                  )
+              )
+            : list;
```
[the remark this answers is not in the document]
Comment (K33k0) [reply lib/routes/vice/topic.ts]: I've had to add a little extra as i was getting a couple errors. However I couldn't have done it without the exampe set here. Thanks, I've included your name in the commit message

File: lib/routes/vice/topic.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                                            alt: component.alt,
+                                            caption: component.caption,
+                                        },
+                                    });
+                                case 'oembed':
+                                    return render({ oembed: { html: component.oembed.html } });
+                                case 'tweet':
+                                    return render({ oembed: { html: component.oembed.html } });
+                                case 'youtube':
+                                    return render({ oembed: { html: component.oembed.html } });
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:88]:
```suggestion
                                case 'tweet':
                                case 'youtube':
                                    return render({ oembed: { html: component.oembed.html } });
```

File: lib/routes/vice/topic.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                                case 'article':
+                                    return render({ article: { html: component.article.body } });
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:90]: `article` role provide some relative articles of the current article. I personally don't like having those's full articles embeded in the current article.

File: lib/routes/vice/topic.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                        .join('');
+
+                return item;
+            })
+        )
+    );
+
+    return {
+        // channel title
+        title: `Vice.com | ${topic} articles`,
```
Comment (TonyRL) [inline lib/routes/vice/topic.ts:104]:
```suggestion
        title: `VICE | ${topic} articles`,
```

File: lib/routes/vice/namespace.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import type { Namespace } from '@/types';
+
+export const namespace: Namespace = {
+    name: 'vice',
```
Comment (TonyRL) [inline lib/routes/vice/namespace.ts:4]:
```suggestion
    name: 'VICE',
```

File: lib/routes/vice/topic.ts (hunk 7 above)
Comment (K33k0) [reply lib/routes/vice/topic.ts]: I see what you are saying with this one. I was mainly doing it for news stories that essentially just contained a list of other articles. I didn't realise the mess it was making inside the other articles. I have now removed it in my latest commit. Thanks again for the pointers.

**Pull request merged**